src/btree/bt_cursor.c

```diff
@@ -682,22 +662,24 @@ __wt_btcur_insert(WT_CURSOR_BTREE *cbt)
 			goto done;
 
 		/*
-		 * The pinned page goes away if we fail for any reason, make
-		 * sure there's a local copy of any key. (Restart could still
-		 * use the pinned page, but that's an unlikely path.) Re-save
-		 * the cursor state: we may retry but eventually fail.
+		 * The pinned page goes away if we fail for any reason, ensure
+		 * there's a local copy of any key or value. (Restart could
+		 * still use the pinned page, but that's an unlikely path.)
+		 * Re-save the cursor state: we may retry but eventually fail.
 		 */
-		WT_TRET(__cursor_copy_int_key(cursor));
+		WT_TRET(__cursor_localkey(cursor));
+		WT_TRET(__cursor_localvalue(cursor));
```

Comment:
This case seems fine to me: it will be very unusual to have an internal value here, so normal cases won't involve any copying.